### ISSUE TITLE
[FEA]: Remove support for CDPv1 in Thrust algorithms 

### ISSUE THREAD

**Issue by jrhemstad** [NVIDIA/cccl#1138]
### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this request and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Area

Thrust

### Is your feature request related to a problem? Please describe.

As described in https://github.com/NVIDIA/cccl/issues/1137, CDPv2 removed the ability to do device-side synchronization. As a result, Thrust algorithms invoked in device code with the `thrust::device` execution policy implicitly fall back to using `thrust::seq`. 

However, Thrust preserved the ability to still use CDPv1 on architectures/CTKs where it is still supported. This has proven to be costly to preserve and deemed no longer worth the effort. 

In other words, this kernel has different behavior depending on how it is compiled:
```
__global__ void kernel(){
   thrust::reduce(thrust::device, ...);
}
```

1. Launches a kernel (CDPv1)
   - Only if `-rdc=true -arch=sm_N` for `N < 90` and CTK < 12.0

2. Falls back to `thrust::seq` 
    - If `-arch=sm_N` for `N >= 90` or CTK >= 12.0
 
    

### Describe the solution you'd like

Invoking `thrust::algorithm(thrust::device,...)` in device code should always be equivalent to `thrust::allgorithm(thrust::seq,...)` (case 2 above).

**Comment by bernhardmgruber** [comment]
Given support for CTK < 12.0 has been removed, CDPv1 is entirely obsolete now.

**Comment by alliepiper** [comment]
> Given support for CTK < 12.0 has been removed, CDPv1 is entirely obsolete now.

Note that this isn't quite correct, CDPv1 is still available on CTK12.0 when `CUDA_FORCE_CDP1_IF_SUPPORTED` is defined: https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compatibility-and-interoperability

That said, I don't think we really care about that usecase and I'm happy to see this mess disappear forever 😆

**Comment by bernhardmgruber** [comment]
Fortunately, we pulled the plug on that flag with CCCL 3.0: #3344. We error out when `CUDA_FORCE_CDP1_IF_SUPPORTED` is defined:

https://github.com/NVIDIA/cccl/blob/804dd154d05d7fdcf7d401407d014ae7501379fe/cub/cub/detail/detect_cuda_runtime.cuh#L89

**Comment by alliepiper** [comment]
Beautiful 🥲
